internal/cmd/start.go

```diff
@@ -0,0 +1,160 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/spinup-host/api"
+	"github.com/spinup-host/config"
+	"github.com/spinup-host/metrics"
+
+	"github.com/golang-jwt/jwt"
+	"github.com/rs/cors"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	cfgFile string
+	uiPath string
+
+	apiPort = ":4434"
+	uiPort = ":3000"
+)
+
+func apiHandler() http.Handler {
+	rand.Seed(time.Now().UnixNano())
+	mux := http.NewServeMux()
+	mux.HandleFunc("/hello", api.Hello)
+	mux.HandleFunc("/createservice", api.CreateService)
+	mux.HandleFunc("/githubAuth", api.GithubAuth)
+	mux.HandleFunc("/logs", api.Logs)
+	mux.HandleFunc("/jwt", api.JWT)
+	mux.HandleFunc("/jwtdecode", api.JWTDecode)
+	mux.HandleFunc("/streamlogs", api.StreamLogs)
+	mux.HandleFunc("/listcluster", api.ListCluster)
+	mux.HandleFunc("/cluster", api.GetCluster)
+	mux.HandleFunc("/metrics", metrics.HandleMetrics)
+	mux.HandleFunc("/createbackup", api.CreateBackup)
+	c := cors.New(cors.Options{
+		AllowedOrigins: []string{"https://app.spinup.host", "http://localhost:3000"},
+		AllowedHeaders: []string{"authorization", "content-type"},
+	})
+
+	return c.Handler(mux)
+}
+
+func uiHandler() http.Handler {
+	fs := http.FileServer(http.Dir(uiPath))
+	http.Handle("/", fs)
+
+	return http.DefaultServeMux
+}
+
+func startCmd() *cobra.Command {
+	sc := &cobra.Command{
+		Use:   "start",
+		Short: "start the spinup API and frontend servers",
+		Run: func(cmd *cobra.Command, args []string) {
+			home, err := os.UserHomeDir()
+			if err != nil {
+				log.Fatalf("FATAL: obtaining home directory: %v", err)
+			}
+			cmd.PersistentFlags().StringVar(&cfgFile, "config",
+				fmt.Sprintf("%s/.local/spinup/config.yaml", home), "Path to spinup configuration")
+			cmd.PersistentFlags().StringVar(&uiPath, "ui-path",
+				fmt.Sprintf("%s/.local/spinup/spinup-dash", home), "Path to spinup frontend")
+
+			log.Println(fmt.Sprintf("INFO: using config file: %s", cfgFile))
+			if err = validateConfig(cfgFile); err != nil {
+				log.Fatalf("FATAL: validating config: %v", err)
+			}
+			log.Println("INFO: initial validations successful")
+
+			apiListener, err := net.Listen("tcp", apiPort)
```

Comment:
maybe stupid question. isn't it HTTP traffic?

Reply:
Haha right, looks like all subset of "TCP" has to use TCP as the network parameter https://pkg.go.dev/net#Listen